src/main/java/homework_4/singleton/Main.java

```diff
@@ -0,0 +1,18 @@
+package homework_4.singleton;
+
+//Example of using Singleton
+
+public class Main {
+    public static void main(String[] args) {
+
+        Circle myCircle = Circle.getInstance();
+        myCircle.setColor("green");
```

Comment:
Opt: 
Not sure about lombok including necessity here at all.
If you want to show ability to change object state you could realize it youself ;)